### PR TITLE
Add nimble tests and add array assertions

### PR DESCRIPTION
**Pull request by markGilchrist**
OK so we don't have to create two targets as nimble uses the XCTAssertions under the hood.

I have added tests for nimble to prove it works in the example project and added some more tests for the functions 

I have also split the functions into files so they are easier to find

let me know your concerns

### CODE REVIEW

File: Example/Tests/NimbleTests.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                    it("test_assertThatError") {
+                        // arrange + act
+                        Observable
+                            .error(TestError.nullPointerException)
+                            .subscribe(intObserver)
+                            .disposed(by: bag)
+                        
+                        // assert
+                        intObserver.assertThatError(that: { (e) -> Bool in
+                            if let _ = e as? TestError, case _ = TestError.nullPointerException {
```
Comment (andecoder) [inline Example/Tests/NimbleTests.swift]: does this really works as we think? If you test with a new case does it fail?
i.e.: `case _ = TestError.aNewError`

File: RxTestExt/Classes/RxTextExtensions.swift [hunk 2]
```diff
@@ -2,57 +2,46 @@ import RxTest
...
         if let error = getErrorSafely(file: file, line: line) {
             XCTAssertEqual(error.localizedDescription, message, file: file, line: line)
         }
     }
-
+    
     // MARK: internal functions
-
+    
     internal func getElementSafely(_ index: Int, file: StaticString, line: UInt ) -> Element? {
```
Comment (andecoder) [inline RxTestExt/Classes/RxTextExtensions.swift]: I don't think the name of this function makes it clear that the test will fail if couldn't get the element safely.

**Review by andecoder — COMMENTED** [review]
Need some clarification, happy to approve after those are clear

File: Example/Tests/NimbleTests.swift (hunk 1 above)
Comment (markGilchrist) [reply Example/Tests/NimbleTests.swift]: @andecoder thanks for you feed back on this, I have added a failing test and you were right

File: RxTestExt/Classes/RxTextExtensions.swift (hunk 2 above)
Comment (markGilchrist) [reply RxTestExt/Classes/RxTextExtensions.swift]: I have renamed to `getElementOrFail` would you think that was clearer?

File: Example/Tests/Tests.swift [hunk 3]
```diff
@@ -10,12 +11,12 @@ class Tests: XCTestCase {
 
     enum TestError: LocalizedError {
         case nullPointerException
+        case andersonsError
```
Comment (sindhu238) [inline Example/Tests/Tests.swift:14]: Really?

Comment (markGilchrist) [reply Example/Tests/Tests.swift:14]: @sindhu238 would you rather it was named after you?

**Review by andecoder — APPROVED** [review]
LGTM

**Pull request merged**